# **E-commerce Sales Analysis Report**

### 1. Project Introduction
This project focuses on analyzing e-commerce sales data to gain a deeper understanding of business performance. By exploring sales and profit trends across different dimensions, such as time, product categories, and customer segments, this analysis aims to identify key strengths, weaknesses, and opportunities for business growth. The insights gathered can be used to make data-driven decisions to optimize inventory, marketing strategies, and overall profitability.


### 2. Tools Used in Project
The following Python libraries and tools were used for data manipulation, analysis, and visualization:

* **pandas:** For loading, cleaning, and transforming the dataset.
* **plotly.express** and **plotly.graph_objects:** For creating interactive data visualizations such as line charts, bar charts, and pie charts.
* **plotly.io** and **plotly.colors:** To set a consistent theme and color palette for the visualizations.
* **Google Colab:** The environment where the code was executed.

### 3. Dataset Introduction
The analysis was performed on the **Superstore.csv** dataset. The dataset contains 9,994 entries and 21 columns, providing detailed information about sales transactions. Key columns include:
* **Sales:** The total amount of a sale.
* **Profit:** The profit generated from a sale.
* **Order Date:** The date a customer placed an order.
* **Category** and **Sub-Category:** Product classifications.
* **Segment:** The customer segment (Consumer, Corporate, Home Office).
* **Region, State, City:** Geographic information.

### 4. Objective
The primary objectives of this project were to:
* Identify monthly sales and profit trends to understand seasonality.
* Determine the top-performing and underperforming product categories and sub-categories.
* Analyze sales and profit performance across different customer segments.
* Provide actionable recommendations based on the data to enhance business strategy and profitability.


Mounting Google Drive with Google Colab

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Importing Required Libraries for The Project

In [4]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
import plotly.colors as colors
pio.templates.default = "plotly_white"

Loading The Dataset

In [19]:
df = pd.read_csv('/content/drive/MyDrive/Data Science Notebooks/E-commerce Sales Analysis/Superstore.csv', encoding='latin-1')

In [20]:
df.head()

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

Discriptive Statistics

In [21]:
df.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

Information of The Dataset

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

When you load a CSV, date columns are usually read as object (string) types. If they stay as plain text, you can’t easily filter, sort, or calculate time intervals — they’re just text, not real dates.

### 5. Data Cleaning and Preprocessing
The initial dataset was loaded into a pandas DataFrame. The following data cleaning and preprocessing steps were performed:


5.1.  **Date Conversion:**

The `Order Date` and `Ship Date` columns, which were initially in a string format (`object`), were converted to the proper `datetime` format. This step is crucial for performing time-based analysis.

In [23]:
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9994 non-null   int64         
 1   Order ID       9994 non-null   object        
 2   Order Date     9994 non-null   datetime64[ns]
 3   Ship Date      9994 non-null   datetime64[ns]
 4   Ship Mode      9994 non-null   object        
 5   Customer ID    9994 non-null   object        
 6   Customer Name  9994 non-null   object        
 7   Segment        9994 non-null   object        
 8   Country        9994 non-null   object        
 9   City           9994 non-null   object        
 10  State          9994 non-null   object        
 11  Postal Code    9994 non-null   int64         
 12  Region         9994 non-null   object        
 13  Product ID     9994 non-null   object        
 14  Category       9994 non-null   object        
 15  Sub-Category   9994 n

5.2.  **Feature Engineering:**

New columns were created to extract specific time-based information:
    * `Order_Year`: The year of the order.
    * `Order_Month`: The month number of the order.
    * `Order_Day`: The day of the week for the order.

In [25]:
df['Order_Year'] = df['Order Date'].dt.year
df['Order_Month'] = df['Order Date'].dt.month
df['Order_Day'] = df['Order Date'].dt.dayofweek

In [26]:
df.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

          Category  Sub-Category  \
0        Furniture     Bookcases   
1        Furniture        Chairs   
2  Office Supplies        Labels   
3        Furniture        Tables   
4  Office Supplies       Storage   

                                        Product Name     Sales Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600        2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400        3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200        2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775        5   
4                     Eldon Fold 'N Roll Cart System   22.3680        2   

  Discount    Profit  Order_Year  Order_Month  Order_Day  
0     0.00   41.9136        2016           11          1  
1     0.00  219.5820        2016           11          1  
2     0.00    6.8714        2016            6          6  
3     0.45 -383.0310        2015           10          6  
4     0.20    2.5164        2015           10          6  

[5 rows x 24 columns]

## **Exploratory Data Analysis**
The analysis involved creating several visualizations to understand the data at a high level:
* **Monthly Sales and Profit Analysis:** Line and bar charts were used to visualize sales and profit totals for each month of the year, revealing seasonal patterns.
* **Sales and Profit by Category:** Pie charts were created to show the distribution of sales and profit across the main product categories (Technology, Furniture, Office Supplies).
* **Sales and Profit by Sub-Category:** Bar charts were used to compare sales and profit performance for each individual sub-category, highlighting top performers and areas of concern.
* **Sales and Profit by Customer Segment:** A grouped bar chart was used to compare the sales and profit generated by each customer segment (Consumer, Corporate, Home Office).
* **Sales to Profit Ratio:** A pie chart was used to visualize the efficiency of each customer segment by calculating the ratio of total sales to total profit.

1. Monthly Sales Analysis

In [28]:
sales_by_month = df.groupby('Order_Month')['Sales'].sum().reset_index()
sales_by_month

Order_Month        Sales
0             1   94924.8356
1             2   59751.2514
2             3  205005.4888
3             4  137762.1286
4             5  155028.8117
5             6  152718.6793
6             7  147238.0970
7             8  159044.0630
8             9  307649.9457
9            10  200322.9847
10           11  352461.0710
11           12  325293.5035

In [29]:
fig = px.line(
    sales_by_month,
    x = 'Order_Month',
    y = 'Sales',
    title = 'Monthly Sales Analysis'
)
fig.show()

### **Observasion :**
- Lowest sales are in February.
- Highest sales occur in November.

2. Sales by Category

In [30]:
sales_by_category = df.groupby('Category')['Sales'].sum().reset_index()
sales_by_category

Category        Sales
0        Furniture  741999.7953
1  Office Supplies  719047.0320
2       Technology  836154.0330

In [31]:
fig = px.pie(
    sales_by_category,
    values = 'Sales',
    names = 'Category',
    hole = 0.5,
    color_discrete_sequence=px.colors.qualitative.Pastel,
)
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text = 'Sales Analysis by Category', title_font = dict(size=20))

fig.show()

### **Observation**:

- Technology has the highest sales at 836,154, making it the top-performing category.
- Furniture comes next with 741,999 in sales.
- Office Supplies has the lowest sales at 719,047 among the three main categories.

3. Sales Analysis by Sub Category

In [32]:
sales_by_subcategory = df.groupby('Sub-Category')['Sales'].sum().reset_index()
sales_by_subcategory

Sub-Category        Sales
0   Accessories  167380.3180
1    Appliances  107532.1610
2           Art   27118.7920
3       Binders  203412.7330
4     Bookcases  114879.9963
5        Chairs  328449.1030
6       Copiers  149528.0300
7     Envelopes   16476.4020
8     Fasteners    3024.2800
9   Furnishings   91705.1640
10       Labels   12486.3120
11     Machines  189238.6310
12        Paper   78479.2060
13       Phones  330007.0540
14      Storage  223843.6080
15     Supplies   46673.5380
16       Tables  206965.5320

In [34]:
fig = px.bar(
    sales_by_subcategory,
    x = 'Sub-Category',
    y = 'Sales',
    title = 'Sales Analysis by Sub-Category',
)
fig.show()

##**Observation:**

- Phones and Chairs are the top-selling sub-categories, with sales over 330,000 and 328,000 respectively.
- Fasteners have the lowest sales, at just 3,024, making it the least contributing sub-category.

4. Monthly Profit

In [35]:
monthly_profit = df.groupby('Order_Month')['Profit'].sum().reset_index()
monthly_profit

Order_Month      Profit
0             1   9134.4461
1             2  10294.6107
2             3  28594.6872
3             4  11587.4363
4             5  22411.3078
5             6  21285.7954
6             7  13832.6648
7             8  21776.9384
8             9  36857.4753
9            10  31784.0413
10           11  35468.4265
11           12  43369.1919

In [36]:
fig = px.bar(
    monthly_profit,
    x = 'Order_Month',
    y = 'Profit',
    title = 'Monthly Profit',
)
fig.show()

### **Observation:**

- Lowest profit is in January (9,134) and February (10,294), showing a slow start to the year.
- Highest profit is in December (43,369), indicating strong profitability at year-end — likely due to peak sales season.

5. Profit Analysis by Category

In [37]:
profit_by_category = df.groupby('Category')['Profit'].sum().reset_index()
profit_by_category

Category       Profit
0        Furniture   18451.2728
1  Office Supplies  122490.8008
2       Technology  145454.9481

In [38]:
fig = px.pie(
    profit_by_category,
    values = 'Profit',
    names = 'Category',
    hole = 0.5,
    color_discrete_sequence=px.colors.qualitative.Pastel,
)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text = 'Profit Analysis by Category', title_font = dict(size=20))

fig.show()

### **Observation:**

- Technology generates the highest profit (145,455), making it the most profitable category overall.
- Office Supplies follows with 122,491 in profit.
- Furniture shows the lowest profit (18,451), despite having significant sales.

6. Profit by Sub Category

In [39]:
profit_by_subcategory = df.groupby('Sub-Category')['Profit'].sum().reset_index()
profit_by_subcategory

Sub-Category      Profit
0   Accessories  41936.6357
1    Appliances  18138.0054
2           Art   6527.7870
3       Binders  30221.7633
4     Bookcases  -3472.5560
5        Chairs  26590.1663
6       Copiers  55617.8249
7     Envelopes   6964.1767
8     Fasteners    949.5182
9   Furnishings  13059.1436
10       Labels   5546.2540
11     Machines   3384.7569
12        Paper  34053.5693
13       Phones  44515.7306
14      Storage  21278.8264
15     Supplies  -1189.0995
16       Tables -17725.4811

In [40]:
fig = px.bar(
    profit_by_subcategory,
    x = 'Sub-Category',
    y = 'Profit',
    title = 'Profit by Sub-Category',
)
fig.show()

### **Observation:**

- Copiers (55,618) and Phones (44,516) are the most profitable sub-categories, showing strong margins.
- Tables (–17,725) and Bookcases (–3,473) have negative profit, indicating a loss in these areas.
- Supplies also shows a small loss (–1,189).

7. Analyze the sales and profit by Customer Segment

In [41]:
sales_profit_by_segment = df.groupby('Segment').agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()
sales_profit_by_segment

Segment         Sales       Profit
0     Consumer  1.161401e+06  134119.2092
1    Corporate  7.061464e+05   91979.1340
2  Home Office  4.296531e+05   60298.6785

In [44]:
fig = go.Figure()
color_palette = colors.qualitative.Pastel
fig.add_trace(go.Bar(
    x = sales_profit_by_segment['Segment'],
    y = sales_profit_by_segment['Sales'],
    name = 'Sales',
    marker_color = color_palette[0],
))
fig.add_trace(go.Bar(
    x = sales_profit_by_segment['Segment'],
    y = sales_profit_by_segment['Profit'],
    name = 'Profit',
    marker_color = color_palette[1],
))

fig.update_layout(
    title = 'Sales and Profit by Customer Segment',
    xaxis_title = 'Customer Segment',
    yaxis_title = 'Amount',
    barmode = 'group',
    legend_title = 'Metrics',
)
fig.show()

### **Observation:**

- The Consumer segment has the highest sales (1,161,401) and also brings in the highest profit (134,119).
- The Corporate segment ranks second for both sales (706,146) and profit (91,979).
- The Home Office segment has the lowest sales (429,653) and the lowest profit (60,299) among the three.

8. Sales to Profit Ratio

In [45]:
sales_profit_by_segment = df.groupby('Segment').agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()
sales_profit_by_segment['Sales_to_Profit_Ratio'] = sales_profit_by_segment['Sales'] / sales_profit_by_segment['Profit']
sales_profit_by_segment

Segment         Sales       Profit  Sales_to_Profit_Ratio
0     Consumer  1.161401e+06  134119.2092               8.659471
1    Corporate  7.061464e+05   91979.1340               7.677245
2  Home Office  4.296531e+05   60298.6785               7.125416

In [48]:
fig = px.pie(
    sales_profit_by_segment,
    values = 'Sales_to_Profit_Ratio',
    names = 'Segment',
    hole = 0.5,
    color_discrete_sequence=px.colors.qualitative.Pastel,
    title= 'Sales to Profit Ratio',
)
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()


### **Observation:**

- Consumer segment has the highest ratio (8.66), meaning it generates about ₹8.66 in sales for every ₹1 of profit.
- Corporate segment follows with a ratio of 7.68.
- Home Office has the lowest ratio (7.13), showing it’s relatively more profitable per unit of sales than the Consumer segment.

### 7. Key Findings
Based on the exploratory data analysis, the following key findings were identified:

* **Seasonal Trends:** The lowest sales and profit occur at the beginning of the year, particularly in **February** and **January**, respectively. Sales and profit peak toward the end of the year, with **November** having the highest sales and **December** having the highest profit.
* **Category Performance:**
    * **Technology** is the top-performing category, generating the highest sales ($836,154) and profit ($145,455).
    * **Furniture** has the second-highest sales ($741,999) but the lowest profit ($18,451).
    * **Office Supplies** has the lowest sales ($719,047) but the second-highest profit ($122,491), indicating strong margins.
* **Sub-Category Performance:**
    * **Phones** and **Copiers** are the most profitable sub-categories.
    * **Tables**, **Bookcases**, and **Supplies** are consistently unprofitable, with Tables showing the most significant loss (-$17,725).
* **Customer Segment Performance:**
    * The **Consumer** segment is the largest driver of both sales and profit.
    * The **Home Office** segment, while having the lowest overall sales and profit, has the best sales-to-profit ratio (7.13), indicating a more efficient conversion of sales into profit.

### 8. Recommendations
Based on the key findings, the following recommendations are suggested to improve business strategy and profitability:

1.  **Address Underperforming Products:** Conduct a detailed cost analysis for the **Tables** and **Bookcases** sub-categories to identify the root cause of their unprofitability. Consider adjusting pricing, negotiating with suppliers, or removing these items from the product catalog if they cannot be made profitable.
2.  **Optimize Seasonal Strategy:** Develop targeted marketing campaigns for the low-sales months, such as **February**, to drive sales. Conversely, maximize marketing and inventory efforts in high-sales months like **November** and **December** to capitalize on peak demand.
3.  **Capitalize on the Home Office Segment:** The high profitability ratio of the Home Office segment presents a significant growth opportunity. Develop marketing campaigns and promotions specifically for this customer group, focusing on high-margin products from the **Technology** and **Office Supplies** categories to further boost their profitability.

### 9. Conclusion
This e-commerce sales analysis provides a clear, data-driven overview of the business's performance. By understanding which products, months, and customer segments drive sales and profit, the business can take strategic action to optimize its operations. Focusing on improving the profitability of key categories and leveraging strengths in high-margin segments are the most impactful next steps for a sustainable and profitable future.